### ISSUE TITLE
False negative for callback protocols that accept `*args`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4414]
The following should generate an error.

```python
from typing import Any, Protocol

class Proto(Protocol):
    def __call__(self, x: int, *args: Any) -> int:
        return 0

def func1(x: int, s: str) -> int:
    return 0

x: Proto = func1
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.288, which I just published. It will also be included in a future release of pylance.
